pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotInstanceRestletResource.java

```diff
@@ -123,10 +123,8 @@ public String getInstance(
   })
   public SuccessResponse addInstance(Instance instance) {
     LOGGER.info("Instance creation request received for instance " + instance.toInstanceId());
-    final PinotResourceManagerResponse resp = pinotHelixResourceManager.addInstance(instance);
-    if (resp.status == PinotResourceManagerResponse.ResponseStatus.failure) {
-      throw new ControllerApplicationException(LOGGER, "Instance already exists",
-          Response.Status.CONFLICT);
+    if (!pinotHelixResourceManager.addInstance(instance).isSuccessful()) {
+      throw new ControllerApplicationException(LOGGER, "Instance already exists", Response.Status.CONFLICT);
```

Comment:
Can you include the instance name or some form of identifying it in the message?

Reply:
In this PR, I kept the response message the same. Will have a separate PR for fixing (adding details into) response message.